pkg/apis/kops/util/labels_test.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestGetNodeRole(t *testing.T) {
+	tests := []struct {
+		name     string
+		node     *v1.Node
+		expected string
+	}{
+		{
+			name: "RoleNone",
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node-1",
+				},
+			},
+			expected: "",
+		},
+		{
+			name: "RoleNewerLabel",
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node-2",
+					Labels: map[string]string{
+						"node-role.kubernetes.io/master": "node-role",
+						"node-role.kubernetes.io/slave":  "node-role",
```

Comment:
nit: can we use `node` instead of `slave` here? Kops uses `node` so it probably makes sense to match that.

https://github.com/kubernetes/kops/blob/942c8915db30780e4b1686218454e0afc2b73df2/pkg/nodelabels/builder.go#L31